File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    
    heapq.heapify(scoville)
    
    while scoville[0] < K :
        if len(scoville) == 1 :
            return -1

        answer += 1
        
        new_scoville = heapq.heappop(scoville) + heapq.heappop(scoville) * 2
        heapq.heappush(scoville, new_scoville)
        
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int s : scoville) {
            pq.offer(s);
        }
        // System.out.println(pq);
        
        while (true) {
            int min_value = pq.peek(); 
            
            if (min_value >= K) {
                break;
            }
            
            // 2개 안남음
            if (pq.size() < 2) {
                answer = -1;
                break;
            }
            
            int min1 = pq.poll();
            int min2 = pq.poll();
            
            int new_scoville = min1 + min2 * 2;
            pq.offer(new_scoville);
            answer++;
        }
        
        
        return answer;
    }
}